src/activity_handlers.cpp

```diff
@@ -474,417 +474,131 @@ int butcher_time_to_cut( const player &u, const item &corpse_item, const butcher
+        } else if( mt.has_flag( MF_HUMAN ) ) {
+            harvest_id_name = "human";
+        } else if( mt.size >= MS_SMALL && mt.size <= MS_MEDIUM ) {
+            if( mt.has_flag( MF_LEATHER ) ) {
+                harvest_id_name = "mammal_leather";
+            } else if( mt.has_flag( MF_FUR ) ) {
+                harvest_id_name = "mammal_fur";
+            } else if( mt.has_flag( MF_WOOL ) ) {
+                harvest_id_name = "mammal_wool";
+            } else if( mt.has_flag( MF_FEATHER ) ) {
+                harvest_id_name = "bird_small";
+            } else if( mt.has_flag( MF_AQUATIC ) ) {
+                harvest_id_name = "fish_large";
+            } else if( mt.has_flag( MF_CHITIN ) ) {
+                harvest_id_name = "arachnid";
+            } else if( mt.has_flag( MF_BONES ) ) {
+                harvest_id_name = "animal_noskin";
+            } else {
+                harvest_id_name = "meatslug";
             }
-            if( parts.charges > 0 ) {
-                g->m.add_item_or_charges( p->pos(), parts );
+        } else if( mt.size >= MS_LARGE ) {
+            if( mt.has_flag( MF_LEATHER ) ) {
+                harvest_id_name = "mammal_large_leather";
+            } else if( mt.has_flag( MF_FUR ) ) {
+                harvest_id_name = "mammal_large_fur";
+            } else if( mt.has_flag( MF_WOOL ) ) {
+                harvest_id_name = "mammal_large_wool";
+            } else if( mt.has_flag( MF_FEATHER ) ) {
+                harvest_id_name = "bird_large";
+            } else if( mt.has_flag( MF_AQUATIC ) ) {
+                harvest_id_name = "fish_large";
+            } else if( mt.has_flag( MF_CHITIN ) ) {
+                harvest_id_name = "arachnid";
+            } else if( mt.has_flag( MF_BONES ) ) {
+                harvest_id_name = "animal_large_noskin";
+            } else {
+                harvest_id_name = "meatslug";
             }
         }
     }
+
+    return harvest_list::all().find( harvest_id( harvest_id_name ) )->second;
 }
 
-void butchery_drops_harvest( item *corpse_item, const mtype &mt, player &p, const time_point &age,
-                             const std::function<int()> &roll_butchery, butcher_type action )
+void butchery_drops_harvest( item *corpse_item, const mtype &mt, player &p, const time_point &bday,
+                             const std::function<int()> &roll_butchery, butcher_type action,
+                             const std::function<double()> &roll_drops )
 {
     p.add_msg_if_player( m_neutral, _( mt.harvest->message().c_str() ) );
+    int monster_weight = to_gram( mt.weight );
+    monster_weight -= monster_weight * rng_float( -0.1, 0.1 );
+    if( corpse_item->has_flag( "QUARTERED" ) ) {
+        monster_weight = monster_weight / 4;
+    }
 
+    int monster_weight_remaining = monster_weight;
     int practice = 4 + roll_butchery();
-    for( const auto &entry : *mt.harvest ) {
+
+    harvest_list harvest_iterator = *mt.harvest;
```

Comment:
I don't think this is an iterator in the C++ sense, right?  In which case this may not be a good name for this variable.  Can you call it just `harvest`?